packages/models/src/resource-binding/manager.ts

```diff
@@ -28,43 +36,65 @@ export class Manager extends Actor<object, MessagePayload<ResourceBindingManager
   }
 
   private update(pollingInterval = 1000) {
+    let semaphore = 0
     return setInterval(async () => {
-      if (this.#tipBlockNumber.gt(0) && this.#tipBlockNumber.gt(BI.from(this.#lastBlock?.header.number ?? 0))) {
-        this.updateStore(await this._dataSource.getBlock(this.#tipBlockNumber.toHexString()))
+      semaphore++
+      if (semaphore === 1) {
+        if (this.#buffer.hasReadyStore()) {
+          this.updateBuffer()
+        }
+        if (this.#tipBlockNumber.gt(0) && this.#tipBlockNumber.gt(BI.from(this.#lastBlock?.header.number ?? 0))) {
+          this.updateStore(await this._dataSource.getBlock(this.#tipBlockNumber.toHexString()))
```

Comment:
Will `await this._dataSource.getBlock(this.#tipBlockNumber.toHexString())` be `undefined` or throw an error?